fv3core/stencils/d_sw.py

```diff
@@ -828,8 +828,8 @@ def d_sw(
         origin=grid().compute_origin(),
         domain=grid().domain_shape_compute(add=(1, 1, 0)),
     )
-
-    ytp_v.compute(vb, v, ub)
+    ytp_v_obj = ytp_v.YTP_V(spec.namelist)
```

Comment:
my understanding here is that `d_sw()` is called in it's `compute` function which is part of the main loop.
Why do we initialize the object in here? This leads to significant overhead as compared to initializing it once with the namelist and then continuously "only" calling it?

Reply:
We do this as an intermediate step, and when D_SW is an object, you are right, this part will happen in its __init__ routine. We've added the ability to memoize class instantiation -- I meant to add it here thanks for the reminder! (first version you just had to add the decorator to the class for it to work, now we decide to do it directly)

Reply:
and where would the DSW object be instantiated?

Reply:
in the DynCore object when it is made, until it is made could be memoized init in the dyn_core compute method